### PR TITLE
Changes separated from Mat 1D support in core #18594

### PR DESCRIPTION
**Pull request by fengyuentau**
Changes separated from #18594

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/core/test/test_mat.cpp [hunk 1]
```diff
@@ -2354,6 +2354,95 @@ TEST(Mat, regression_18473)
     EXPECT_EQ((int)5, (int)m.at<short>(19, 49, 99));
 }
 
+// TEST(Mat1D, basic)
```
Comment (fengyuentau) [inline modules/core/test/test_mat.cpp:2357]: In the test of core module, there is no skip flag like the one in dnn tests. So comment for now.

Comment (asmorkalov) [reply modules/core/test/test_mat.cpp:2357]: You can add DISABLED_ to test name. It's compiled and skipped during default test execution.

Comment (fengyuentau) [reply modules/core/test/test_mat.cpp:2357]: Thank you! Let me have a try.

Comment (asmorkalov) [reply modules/core/test/test_mat.cpp:2357]: https://github.com/google/googletest/blob/main/docs/advanced.md#temporarily-disabling-tests

**Review by vpisarev — APPROVED** [review]

File: modules/core/src/minmax.cpp [hunk 2]
```diff
@@ -1565,13 +1565,24 @@ void cv::minMaxLoc( InputArray _img, double* minVal, double* maxVal,
 {
     CV_INSTRUMENT_REGION();
 
-    CV_Assert(_img.dims() <= 2);
+    int dims = _img.dims();
+    CV_Assert(dims <= 2);
```
Comment (alalek) [inline modules/core/src/minmax.cpp:1569]: `CV_CheckLE(dims, 2, "")` is preferable.

File: modules/core/test/test_mat.cpp (hunk 1 above)
Comment (fengyuentau) [reply modules/core/test/test_mat.cpp]: Added prefix DISABLED_ to the test.

File: modules/core/src/minmax.cpp (hunk 2 above)
Comment (fengyuentau) [reply modules/core/src/minmax.cpp:1569]: Done. Thank you.

**Review by alalek — APPROVED** [review]
Please squash commits before merge.

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**